internal/mempool/mempool.go

```diff
@@ -1123,7 +1251,8 @@ func (g *txDescGraph) addChild(tx *mining.TxDesc, child *mining.TxDesc) {
 func (g *txDescGraph) addParent(tx *mining.TxDesc, parent *mining.TxDesc) {
 	txHash := *tx.Tx.Hash()
 	if _, exists := g.parentsOf[txHash]; !exists {
-		g.parentsOf[txHash] = make(map[chainhash.Hash]*mining.TxDesc)
+		g.parentsOf[txHash] = make(map[chainhash.Hash]*mining.TxDesc,
+			len(tx.Tx.MsgTx().TxIn))
```

Comment:
This can be the lesser of `len(tx.Tx.MsgTx().TxIn)` and `ancestorTrackingLimit`.